plugins/aws/resoto_plugin_aws/resource/base.py

```diff
@@ -119,6 +120,24 @@ def to_json(self) -> Json:
             ),
         )
 
+    def set_arn(
+        self,
+        builder: GraphBuilder,
+        region: Optional[str] = None,
+        service: Optional[str] = None,
+        account: Optional[str] = None,
+        resource: str = "",
```

Comment:
According to https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arns-paths there is also an optional resource type. Consider to add.

Reply:
For everything after the account it, there's a lot of variation. Can be either of
```
arn:partition:service:region:account-id:resource-id
arn:partition:service:region:account-id:resource-type/resource-id
arn:partition:service:region:account-id:resource-type:resource-id
```
The method as written is bundling this under `resource`, because you'd need a whole lot of context otherwise. See for example:
```py
deploy_instance.set_arn(
    builder=builder,
    account="",
    resource=f"/restapis/{api_instance.id}/deployments/{deploy_instance.id}",
)
```